### PR TITLE
Pp 446 upgrade notify

### PR DESCRIPTION
**Pull request by mrlumbu**
Upgrade to use new GOV.UK Notify library.

Main advantage of this is it does not use it's own, unconfigurable HTTP client, but just uses URL, so picks up proxy settings which we need for it to talk out through egress proxy.

Off the back of this some refactoring was needed in UserNotificationService and tests.

Currently UserNotificationIntTests are disabled as the new Notify client casts all urls to https, making it very hard to test locally. This will be fixed by submitting PR to Notify.

### CODE REVIEW

**Comment by mrlumbu** [comment]
rebuild

**Comment by mrlumbu** [comment]
rebuild

File: src/main/java/uk/gov/pay/connector/service/UserNotificationService.java [hunk 1]
```diff
@@ -85,7 +73,7 @@ private void readEmailConfig(ConnectorConfiguration configuration) {
         }
     }
 
-    private Map<String, String> buildEmailPersonalisationFromCharge(ChargeEntity charge) {
+    private HashMap<String, String> buildEmailPersonalisationFromCharge(ChargeEntity charge) {
```
Comment (PauloPortugal) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: Shouldn't we be coding for interfaces?

File: src/main/java/uk/gov/pay/connector/service/UserNotificationService.java [hunk 2]
```diff
@@ -94,14 +82,16 @@ private void readEmailConfig(ConnectorConfiguration configuration) {
...
 
-        return new ImmutableMap.Builder<String, String>()
-                .put("serviceReference", charge.getReference())
-                .put("date", DateTimeUtils.toUserFriendlyDate(charge.getCreatedDate()))
-                .put("amount", formatToPounds(charge.getAmount()))
-                .put("description", charge.getDescription())
-                .put("customParagraph", StringUtils.defaultString(customParagraph))
-                .put("serviceName",  StringUtils.defaultString(gatewayAccount.getServiceName()))
-                .build();
+        HashMap<String, String> map = new HashMap<>();
```
Comment (PauloPortugal) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: Same thing here. 
Why is an ImutableMap not suitable for this scenario?

File: src/main/java/uk/gov/pay/connector/service/UserNotificationService.java (hunk 1 above)
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: We should, but unfortunately the NotifyClient specifies HashMap

File: src/main/java/uk/gov/pay/connector/service/UserNotificationService.java (hunk 2 above)
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: See above

File: src/test/java/uk/gov/pay/connector/unit/service/UserNotificationServiceTest.java [hunk 3]
```diff
@@ -58,22 +54,42 @@ public void setUp() {
     @Test
     public void shouldSendEmail() throws Exception {
         when(mockNotifyClientProvider.get()).thenReturn(mockNotifyClient);
-        when(mockNotifyClient.sendEmail(any(EmailRequest.class))).thenReturn(mockNotificationCreatedResponse);
-        when(mockNotificationCreatedResponse.getId()).thenReturn("100");
+        when(mockNotifyClient.sendEmail(any(), any(), any())).thenReturn(mockNotificationCreatedResponse);
```
Comment (rahul-vinayak) [inline src/test/java/uk/gov/pay/connector/unit/service/UserNotificationServiceTest.java]: better to use any(SpecificClass.class) than just any,

File: src/test/java/uk/gov/pay/connector/unit/service/UserNotificationServiceTest.java [hunk 4]
```diff
@@ -87,85 +103,16 @@ public void testEmailSendingThrowsExceptionForMissingTemplate() throws Exception
...
-        verify(mockNotifyClient).sendEmail(argThat(new IsEmailRequestWithParameters(expectedParameters)));
-    }
 
     @Test
     public void testEmailSendWhenEmailsNotifyDisabled() throws Exception {
         when(mockNotifyConfiguration.isEmailNotifyEnabled()).thenReturn(false);
         when(mockNotifyClientProvider.get()).thenReturn(mockNotifyClient);
-        when(mockNotifyClient.sendEmail(any(EmailRequest.class))).thenReturn(mockNotificationCreatedResponse);
-        when(mockNotificationCreatedResponse.getId()).thenReturn("100");
+        when(mockNotifyClient.sendEmail(any(), any(), any())).thenReturn(mockNotificationCreatedResponse);
```
Comment (rahul-vinayak) [inline src/test/java/uk/gov/pay/connector/unit/service/UserNotificationServiceTest.java]: any(SpecificClass.class)

File: src/main/java/uk/gov/pay/connector/service/UserNotificationService.java (hunk 1 above)
Comment (PauloPortugal) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: 0_o

Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: 😢

**Comment by mrlumbu** [comment]
rebuild

**Comment by PauloPortugal** [comment]
rebuild

**Comment by mrlumbu** [comment]
rebuild

**Pull request merged**